src/main/java/be/seeseemelk/mockbukkit/entity/HumanEntityMock.java

```diff
@@ -1,24 +1,140 @@
 package be.seeseemelk.mockbukkit.entity;
 
+import be.seeseemelk.mockbukkit.AsyncCatcher;
 import be.seeseemelk.mockbukkit.ServerMock;
+import be.seeseemelk.mockbukkit.UnimplementedOperationException;
 import be.seeseemelk.mockbukkit.WorldMock;
+import be.seeseemelk.mockbukkit.inventory.InventoryMock;
+import be.seeseemelk.mockbukkit.inventory.PlayerInventoryMock;
+import be.seeseemelk.mockbukkit.inventory.PlayerInventoryViewMock;
+import be.seeseemelk.mockbukkit.inventory.SimpleInventoryViewMock;
+import com.google.common.base.Preconditions;
+import org.bukkit.Bukkit;
+import org.bukkit.GameMode;
 import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.block.Sign;
+import org.bukkit.entity.Entity;
 import org.bukkit.entity.HumanEntity;
+import org.bukkit.entity.Villager;
+import org.bukkit.event.inventory.InventoryCloseEvent;
+import org.bukkit.event.inventory.InventoryType;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.InventoryView;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.Merchant;
+import org.bukkit.inventory.PlayerInventory;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
 import java.util.UUID;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
 public abstract class HumanEntityMock extends LivingEntityMock implements HumanEntity
 {
 
+	private @Nullable PlayerInventoryMock inventory = null;
+	private InventoryView inventoryView;
+	private @Nullable ItemStack cursor = null;
+	private @NotNull GameMode gameMode = GameMode.SURVIVAL;
 	private @Nullable Location lastDeathLocation = new Location(new WorldMock(), 0, 0, 0);
+	protected int expLevel = 0;
+	private float saturation = 5.0F;
+	private int foodLevel = 20;
+
+
 
 	protected HumanEntityMock(@NotNull ServerMock server, @NotNull UUID uuid)
 	{
 		super(server, uuid);
 	}
 
+	/**
+	 * Assert that the player is in a specific gamemode.
+	 *
+	 * @param expectedGamemode The gamemode the player should be in.
+	 */
+	public void assertGameMode(GameMode expectedGamemode)
+	{
+		assertEquals(expectedGamemode, gameMode);
+	}
+
+	@Override
+	public @NotNull PlayerInventory getInventory()
+	{
+		if (inventory == null)
+		{
+			inventory = (PlayerInventoryMock) Bukkit.createInventory(this, InventoryType.PLAYER);
```

Comment:
?

Reply:
Whats the Issue here. Thats how it was implemented in PlayerMock